OneCommunityGlobal/HGNRest
# Sav refactor userHelper.js

# Description
This is a basic refactoring of the userHelper file. This file contains the weekly badge functions and a few other things. I've updated some unused variables, undefined values, and a bunch of hard-to-read loops. To test this PR you can run a test user through the badge functions at the bottom of the file.


## Related PRS (if any):
This is related to all badge bugs but doesn't address any directly.
…


## How to test:
1. check into the current branch.
2. do `npm install` and `...` to run this PR locally.
3. Inside userHelper.js on line 1379 add a filter to select your test user (recommend using firstName).
4. shift awardNewBadges function call on line 16 of the userProfileJobs down to line 17 outside of the Sunday conditional.
5. Change the cronjob syntax to a good frequency for testing on line 8 of userProfileJobs (recommend using '*/2 * * * * ').
6. Add a few console logs to check that the awardNewBadges is running properly.
7. run 'npm run build' and then 'npm start'.
8. observe results and keep an eye out for errors generated.
